### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by najibullahjafari**
Here:
- I added RSpec as a gem to my project.
-  I created a Request spec file for all my controllers.
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    end
+  end
+
+  describe 'GET #show' do
+    let(:post) { Post.create!(title: 'Test Post', text: 'This is a test post.', author: user) }
+
+    it 'assigns @user and @post' do
+      get :show, params: { user_id: user.id, id: post.id }
+      expect(assigns(:user)).to eq(user)
+      expect(assigns(:post)).to eq(post)
```
Comment (Davidosky007) [inline spec/requests/posts_controller_spec.rb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.) that 👇 
```
Make sure that for each action you check:
If the response status was correct.
If a correct template was rendered.
If the response body includes the correct placeholder text.
```

- * Kindly ensure you write at least 3 tests that cover all scenarios stated above for each action

File: spec/requests/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      expect(response.body).to include(other_user.bio)
+    end
+  end
+
+  describe 'Users GET /users#index' do
+    it 'checks if response status was correct' do
+      get users_path, headers: { 'HTTP_ACCEPT' => 'application/json' }
+
+      expect(response).to have_http_status(:success)
+    end
```
Comment (Davidosky007) [inline spec/requests/users_controller_spec.rb]: * Same thing here also 
```
Make sure that for each action you check:
If the response status was correct.
If a correct template was rendered.
If the response body includes the correct placeholder text.
```
* Kindly ensure you write at least 3 tests that cover all scenarios stated above for each action

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @najibullahjafari 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Tests for all controllers written 👍 
* Well-written PR 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @najibullahjafari, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] All controller specs pass with green check marks :white_check_mark: 
- [x] Rspec testing gem is used to test the application :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:     

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**